extraTs/utils/interfaces.ts

```diff
@@ -0,0 +1,43 @@
+import {
+	getUsers,
+	getPasswords,
+	getBranchName,
+	getEmployeeId,
+	getEmployeeAddress
+} from "../constants";
+export interface IPermissions {
+	getUsers: {
+		read: string[];
+		write: string[];
+		delete: string[];
+		all: string[];
+	};
+	getPasswords: {
+		read: string[];
+		write: string[];
+		delete: string[];
+		all: string[];
+	};
+	getEmployeeId: {
+		read: string[];
+		write: string[];
+		delete: string[];
+		all: string[];
+	};
+	getBranchName: {
+		read: string[];
+		write: string[];
+		delete: string[];
+		all: string[];
+	};
+	getEmployeeAddress: {
+		read: string[];
+		write: string[];
+		delete: string[];
+		all: string[];
+	};
+}
+export interface IUsers {
```

Comment:
are the interfaces not redundant you have also defined them in the constants file